src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -2317,19 +2314,15 @@ private void env(String name, String value) {
         else                environment.put(name,value);
     }
 
-    /**
-     * Returns the {@link org.eclipse.jgit.lib.Repository} used by this git instance.
-     *
-     * @return a {@link org.eclipse.jgit.lib.Repository} object.
-     * @throws hudson.plugins.git.GitException if underlying git operation fails.
-     */
-    @NonNull
-    public Repository getRepository() throws GitException {
-        try {
-            return FileRepositoryBuilder.create(new File(workspace, Constants.DOT_GIT));
-        } catch (IOException e) {
-            throw new GitException("Failed to open Git repository " + workspace, e);
-        }
+    @Override
+    public <T> T withRepository(final RepositoryCallback<T> callable) throws IOException, InterruptedException {
+        return workspace.act(new MasterToSlaveFileCallable<T>() {
```

Comment:
stephenc: > I am concerned about the anonymous inner class here that may pull over the outer class reference. Does it make sense to switch to an explicit named static inner class so that we clearly identify the minimal set of class references to send over the remoting reference